### DISCUSSION TITLE
Is there any option for named routes?

### DISCUSSION THREAD

**Discussion by ilyaskarim** [vercel/next.js#68063] [App Router]
Just like named routes in Vue(https://router.vuejs.org/guide/essentials/named-routes.html) do we have option to set named routes in Next.js?

I was building an application where we had dozens of routes and we have to redirect by writing the path name.

I was wondering if we have such option where we can assign path name in Next.js, I'd like something like this.

In route we should be able to do:`app/login/page.js`

```javascript
import LoginPage from "@/components/PageContainers/LoginPage/LoginPage";

import { Metadata } from 'next';


export const RouteMeta = {
    name: 'login',
}

export const metadata: Metadata = {
  title: 'Login - PrApp',
  description: 'Login - PrApp',
};

const LoginPage = () => {
    return <LoginPage></LoginPage>
}

export default LoginPage;

```

So in another file we can make use of this named route like this:

```javascript
<Button
  onClick={() =>
    router.push({
      name: 'login',
    })
  }
>
  Login
</Button>
```

and with `Link`:

```javascript
<Link href={{ name: 'login' }}>
  <a>Login</a>
</Link>
```

Should I open a feature request?

**Comment by pervaizdattoo** [comment]
Yes, we need a feature like named routes in Next.js. I think It would much simplify routing, especially when dealing with numerous paths. Would be great to have something like this out of the box.

**Comment by ilyaskarim** [comment]
Will anybody answer? 🤔
